doc/rbd: rbd encryption (#39478)
ceph/ceph: doc/rbd/rbd-encryption.rst

into the image) and loaded by RBD encryption.

.. note::
   The LUKS formats are supported on Linux-based systems only.

.. note::
   Currently, only AES-128 and AES-256 encryption algorithms are supported.
   Additionally, xts-plain64 is currently the only supported encryption mode.

To use the LUKS format, start by formatting the image::

    $ rbd encryption format {pool-name}/{image-name} {luks1|luks2} {passphrase-file} [–cipher-alg {aes-128 | aes-256}]

The encryption format operation generates a LUKS header and writes it to the
beginning of the image. The header is appended with a single keyslot holding a
randomly-generated encryption key, and is protected by the passphrase read from
`passphrase-file`.

.. note::
   If the content of `passphrase-file` ends with a newline character, it will
   be stripped off.

By default, AES-256 in xts-plain64 mode (which is the current recommended mode,
and the usual default for other tools) will be used. The format operation
allows selecting AES-128 as well. Adding / removing passphrases is currently
not supported by RBD, but can be applied to the raw RBD data using compatible
tools such as cryptsetup.

> The LUKS header size can vary (upto 136MB in LUKS2), but is usually upto 16MB,

Reviewer: ```suggestion
The LUKS header size can vary (up to 136MiB in LUKS2), but is usually defaults to 16MiB,
```

> depending on the version of libcryptsetup installed. For optimal performance,

Reviewer: ```suggestion
depending on the version of `libcryptsetup` installed. For optimal performance,
```

the encryption format will set the data offset to be aligned with the image

> object size. So for example expect a minimum overhead of 8MB if using an image

Reviewer: ```suggestion
object size. For example, expect a minimum overhead of 8MiB if using an image
```

> configured with an 8MB object size.

Reviewer: ```suggestion
configured with an 8MiB object size.
```

In LUKS1, sectors, which are the minimal encryption units, are fixed at 512
bytes. LUKS2 supports larger sectors, and for better performance we set

> the default sector size to the maximum of 4KB. Writes which are either smaller

Reviewer: ```suggestion
the default sector size to the maximum of 4KiB. Writes which are either smaller
```
